src/mon/Monitor.cc

```diff
@@ -306,6 +306,35 @@ void Monitor::do_admin_command(string command, cmdmap_t& cmdmap, string format,
       f->flush(ss);
     }
 
+  } else if (command == "test generate osdmap") {
```

Comment:
this whole thing makes me nervous... it seems inevitable that it will rot.  can we instead do a loop like 'for f in `seq 1 100` ; do ceph osd unset nodown ; done'?

Reply:
i still don't like this patch; i'd rather do it the slow way (for loop in bash doing some trivial change)